docs/superpowers/plans/2026-05-19-waterfall-virtualization-lib-swap.md

```
@@ -0,0 +1,1001 @@
+```
+
+Expected output: a line like `+ @tanstack/react-virtual <X.Y.Z>` and a lockfile update. The version pinned should be the latest stable in the 3.x line (3.10+ as of this writing). Take whatever pnpm resolves.
+
+- [ ] **Step 2: Confirm the dep landed**
+
+```bash
+grep -n "react-virtual" console/web/package.json
+```
+
+Expected output:
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)